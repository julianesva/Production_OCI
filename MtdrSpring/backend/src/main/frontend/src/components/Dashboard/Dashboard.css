.dashboard-main {
    overflow: auto;
    flex: 1;
    height: 100%;
  }

.dashboard-main-container {
    display: flex;
    padding: 20px;
    gap: 20px;
}

.dashboard-loading {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100%;
    width: 100%;
}

.dashboard-error {
    display: flex;
    justify-content: center;
    padding: 20px;
}

.dashboard-error-text {
    font-size: 24px;
    font-weight: bold;
}