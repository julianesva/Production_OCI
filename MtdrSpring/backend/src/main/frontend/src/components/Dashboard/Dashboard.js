import "./Dashboard.css";
import { useState, useEffect } from "react";
import { CircularProgress } from "@mui/material";
import DashboardContent from "./DashboardContent/DashboardContent";
import { API_HEADERS, API_LIST, API_EMPLOYEES, API_MODULES } from "../../API";

export default function Dashboard() {
  const [refresh, setRefresh] = useState(false);
  const [isLoading, setLoading] = useState(false);
  const [isInserting, setInserting] = useState(false);
  const [items, setItems] = useState([]);
  const [employeesList, setEmployeesList] = useState([]);
  const [error, setError] = useState();
  const [modules, setModules] = useState([]);
  const [selectedModule, setSelectedModule] = useState("all");

  function deleteItem(deleteId) {
    fetch(API_LIST + "/" + deleteId, {
      method: "DELETE",
      headers: API_HEADERS,
    })
      .then((response) => {
        if (response.ok) {
          return response;
        } else {
          throw new Error("Something went wrong ...");
        }
      })
      .then(
        (result) => {
          const remainingItems = items.filter((item) => item.id !== deleteId);
          setItems(remainingItems);
        },
        (error) => {
          setError(error);
        }
      );
  }

  function toggleDone(taskData) {
    modifyItem(
      taskData.id,
      taskData.title,
      taskData.description,
      taskData.done,
      taskData.estimatedTime,
      taskData.story_Points,
      taskData.moduleId,
      taskData.actualTime
    ).then(
      (result) => {
        reloadOneIteam(taskData.id);
      },
      (error) => {
        setError(error);
      }
    );
  }

  function modifyItem(
    id,
    title,
    description,
    done,
    estimatedTime,
    story_Points,
    moduleId,
    actualTime
  ) {
    let data = {
      title: title,
      description: description,
      estimatedTime: estimatedTime,
      done: done,
      story_Points: story_Points,
      moduleId: moduleId,
      actualTime: actualTime,
    };
    return fetch(API_LIST + "/" + id, {
      method: "PUT",
      headers: API_HEADERS,
      body: JSON.stringify(data),
    }).then((response) => {
      if (response.ok) {
        return response;
      } else {
        throw new Error("Something went wrong ...");
      }
    });
  }

  function reloadOneIteam(id) {
    fetch(API_LIST + "/" + id, {
      headers: API_HEADERS,
    })
      .then((response) => {
        if (response.ok) {
          return response.json();
        } else {
          throw new Error("Something went wrong ...");
        }
      })
      .then(
        (result) => {
          const items2 = items.map((x) =>
            x.id === id
              ? {
                  ...x,
                  title: result.title,
                  description: result.description,
                  done: result.done,
                  creation_ts: result.creation_ts,
                  estimatedTime: result.estimatedTime,
                  story_Points: result.story_Points,
                  moduleId: result.moduleId,
                  actualTime: result.actualTime,
                }
              : x
          );
          setItems(items2);
        },
        (error) => {
          setError(error);
        }
      );
  }

  function addItem(data) {
    setInserting(true);
    fetch(API_LIST, {
      method: "POST",
      headers: API_HEADERS,
      body: JSON.stringify(data),
    })
      .then((response) => {
        if (response.ok) {
          return response;
        } else {
          throw new Error("Something went wrong ...");
        }
      })
      .then(
        (result) => {
          var id = result.headers.get("location");
          var newItem = {
            id: id,
            title: data.title,
            description: data.description,
            estimatedTime: data.estimatedTime,
            done: data.done,
            story_Points: data.story_Points,
            moduleId: data.moduleId,
            responsible: data.responsible,
          };
          setItems([newItem, ...items]);
          setInserting(false);
        },
        (error) => {
          setInserting(false);
          setError(error);
        }
      );
  }

  function handleModuleChange(event) {
    setSelectedModule(event.target.value);
  }

  useEffect(() => {
    const fetchData = async () => {
      setLoading(true);
      fetch(API_LIST, {
        headers: API_HEADERS,
      })
        .then((response) => {
          if (response.ok) {
            return response.json();
          } else {
            throw new Error("Something went wrong loading Tasks...");
          }
        })
        .then(
          (result) => {
            console.log("API Tasks:", result);
            setItems(result);
          },
          (error) => {
            setError(error);
            setLoading(false);
          }
        );

      fetch(API_EMPLOYEES, {
        headers: API_HEADERS,
      })
        .then((response) => {
          if (response.ok) {
            return response.json();
          } else {
            throw new Error("Something went wrong loading Employees...");
          }
        })
        .then(
          (result) => {
            console.log("API Employees", result);
            setEmployeesList(result);
          },
          (error) => {
            setError(error);
            setLoading(false);
          }
        );

      fetch(API_MODULES, {
        headers: API_HEADERS,
      })
        .then((response) =>
          response.ok
            ? response.json()
            : Promise.reject("Error fetching modules")
        )
        .then((result) => {
          setModules(result);
          setLoading(false);
        })
        .catch((error) => {
          setError(error);
          setLoading(false);
        });
    };

    fetchData();
  }, [refresh]);

  return (
    <div className="dashboard-main">
      {/* Loading OR Dashboard */}
      {isLoading ? (
        // Loading
        <div className="dashboard-loading">
          <CircularProgress />
        </div>
      ) : error ? (
        // Error
        <div className="dashboard-error">
          <p className="dashboard-error-text">Error: {error}</p>
        </div>
      ) : (
        // Dashboard
        <div className="dashboard-main-container">
          <DashboardContent
            items={items}
            employeesList={employeesList}
            addItem={addItem}
            isInserting={isInserting}
            toggleDone={toggleDone}
            deleteItem={deleteItem}
            modules={modules}
            selectedModule={selectedModule}
            handleModuleChange={handleModuleChange}
          />
        </div>
      )}
    </div>
  );
}
