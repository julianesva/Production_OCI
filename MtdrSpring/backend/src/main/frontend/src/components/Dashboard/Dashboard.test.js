import { render, screen, waitFor, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { rest } from "msw";
import { setupServer } from "msw/node";
import Dashboard from "./Dashboard";
import { API_LIST, API_EMPLOYEES, API_MODULES, API_HEADERS } from "../../API";

// Mock data for testing
const mockTasks = [
  {
    id: 1,
    title: "Task 1",
    description: "Description for Task 1",
    estimatedTime: 5,
    done: 0,
    story_Points: 3,
    moduleId: 1,
    responsible: 1,
    actualTime: 0,
  },
  {
    id: 2,
    title: "Task 2",
    description: "Description for Task 2",
    estimatedTime: 8,
    done: 0,
    story_Points: 5,
    moduleId: 2,
    responsible: 2,
    actualTime: 0,
  },
  {
    id: 3,
    title: "Task 3",
    description: "Description for Task 3",
    estimatedTime: 3,
    done: 1,
    story_Points: 2,
    moduleId: 1,
    responsible: 1,
    actualTime: 4,
  },
];

const mockEmployees = [
  {
    id: 1,
    user: {
      id: 1,
      username: "user1",
      email: "user1@example.com",
    },
  },
  {
    id: 2,
    user: {
      id: 2,
      username: "user2",
      email: "user2@example.com",
    },
  },
];

const mockModules = [
  {
    id: 1,
    title: "Sprint 1",
  },
  {
    id: 2,
    title: "Sprint 2",
  },
];

// Setup MSW server for mocking HTTP requests
const server = setupServer(
  // Mock GET request for tasks
  rest.get(`${API_LIST}`, (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(mockTasks));
  }),

  // Mock GET request for employees
  rest.get(API_EMPLOYEES, (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(mockEmployees));
  }),

  // Mock GET request for modules
  rest.get(API_MODULES, (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(mockModules));
  })
);

// Setup and teardown
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

describe("Dashboard Component", () => {
  // Test loading state
  test("shows loading state while fetching data", () => {
    render(<Dashboard />);
    expect(screen.getByRole("progressbar")).toBeInTheDocument();
  });

  // Snapshot test for Dashboard component
  test("Dashboard component matches snapshot", async () => {
    const { container } = render(<Dashboard />);

    // Wait for loading to complete
    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    // Take a snapshot of the rendered component
    expect(container).toMatchSnapshot();
  });

  // Test error handling
  test("shows error message when API request fails", async () => {
    server.use(
      rest.get(`${API_LIST}`, (req, res, ctx) => {
        return res(ctx.status(500));
      })
    );

    render(<Dashboard />);

    await waitFor(() => {
      const errorElement = screen.getByText(
        "Error: Something went wrong loading Tasks..."
      );
      expect(errorElement).toBeInTheDocument();
    });
  });

  // Test successful data loading
  test("loads and displays tasks, employees, and modules data", async () => {
    render(<Dashboard />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    // Check if dashboard title is displayed
    expect(screen.getByText("Dashboard")).toBeInTheDocument();

    // Check if module filter is displayed
    expect(screen.getByText("Filter by Sprint:")).toBeInTheDocument();

    // Check if module options are displayed
    expect(screen.getByText("All")).toBeInTheDocument();
    expect(screen.getByText("1 - Sprint 1")).toBeInTheDocument();
    expect(screen.getByText("2 - Sprint 2")).toBeInTheDocument();
  });

  // Test real-time task visualization Objective number 1
  test("displays tasks assigned to users in real-time", async () => {
    render(<Dashboard />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    // Check if tasks are displayed in the "To Do" table
    expect(screen.getByText("Task 1")).toBeInTheDocument();
    expect(screen.getByText("Task 2")).toBeInTheDocument();

    // Check if tasks are displayed in the "Completed" table
    expect(screen.getByText("Task 3")).toBeInTheDocument();

    // Check if task details are displayed correctly
    expect(screen.getByText("Description for Task 1")).toBeInTheDocument();
  });

  // Test task filtering by module
  test("filters tasks by module when a module is selected", async () => {
    const { container } = render(<Dashboard />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    // Find the filter select element
    const filterContainer = container.querySelector(".filter-module-container");
    const moduleSelect = filterContainer.querySelector("select");
    expect(moduleSelect).toBeInTheDocument();

    // Select module 1
    fireEvent.change(moduleSelect, { target: { value: "1" } });

    // Wait for the UI to update
    await waitFor(() => {
      // Only tasks from module 1 should be visible
      expect(screen.getByText("Task 1")).toBeInTheDocument();
      expect(screen.queryByText("Task 2")).not.toBeInTheDocument();
      expect(screen.getByText("Task 3")).toBeInTheDocument();
    });
  });

  // Test task completion functionality Objective number 3
  test("updates task status when marked as completed", async () => {
    // Mock the PUT request for updating a task
    server.use(
      rest.put(`${API_LIST}/1`, (req, res, ctx) => {
        return res(
          ctx.status(200),
          ctx.json({
            id: 1,
            title: "Task 1",
            description: "Description for Task 1",
            estimatedTime: 5,
            done: 1,
            story_Points: 3,
            moduleId: 1,
            actualTime: 0,
          })
        );
      })
    );

    render(<Dashboard />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    // Find and click the "Done" button for Task 1
    const doneButtons = screen.getAllByText("Done");
    fireEvent.click(doneButtons[0]);

    // Wait for the UI to update
    await waitFor(() => {
      // Task 1 should now be in the "Completed" table
      const completedTasks = screen.getAllByText("Task 1");
      expect(completedTasks.length).toBeGreaterThan(0);
    });
  });

  // Test task filtering by status
  test("filters tasks by status (To Do and Completed)", async () => {
    render(<Dashboard />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    // Check if both tables are displayed
    expect(screen.getByText("To Do")).toBeInTheDocument();
    expect(screen.getByText("Completed")).toBeInTheDocument();

    // Check if tasks are in the correct tables
    expect(screen.getByText("Task 1")).toBeInTheDocument(); // To Do
    expect(screen.getByText("Task 2")).toBeInTheDocument(); // To Do
    expect(screen.getByText("Task 3")).toBeInTheDocument(); // Completed
  });

  // Test task details display Objective number 4
  test("displays task details correctly", async () => {
    render(<Dashboard />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    // Check task details
    expect(screen.getByText("Task 1")).toBeInTheDocument();
    expect(screen.getByText("Description for Task 1")).toBeInTheDocument();
  });

  // Test module filter reset
  test("resets module filter when 'All' is selected", async () => {
    const { container } = render(<Dashboard />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    // Find the filter select element
    const filterContainer = container.querySelector(".filter-module-container");
    const moduleSelect = filterContainer.querySelector("select");

    // Select module 1
    fireEvent.change(moduleSelect, { target: { value: "1" } });

    // Select "All"
    fireEvent.change(moduleSelect, { target: { value: "all" } });

    // Wait for the UI to update
    await waitFor(() => {
      // All tasks should be visible
      expect(screen.getByText("Task 1")).toBeInTheDocument();
      expect(screen.getByText("Task 2")).toBeInTheDocument();
      expect(screen.getByText("Task 3")).toBeInTheDocument();
    });
  });

  // Test task data state changes Objective
  test("displays task data with correct information", async () => {
    render(<Dashboard />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    // Check if task details are displayed correctly in the "To Do" table
    const todoTask = screen.getByText("Task 1").closest("tr");
    expect(todoTask).toHaveTextContent("user1"); // Developer name
    expect(todoTask).toHaveTextContent("3"); // Story points
    expect(todoTask).toHaveTextContent("5"); // Estimated hours

    // Check if task details are displayed correctly in the "Completed" table
    const completedTask = screen.getByText("Task 3").closest("tr");
    expect(completedTask).toHaveTextContent("user1"); // Developer name
    expect(completedTask).toHaveTextContent("2"); // Story points
    expect(completedTask).toHaveTextContent("3"); // Estimated hours
    expect(completedTask).toHaveTextContent("4"); // Actual hours
  });

  // Test changing task data (name, developer, story points, estimated hours) Objective Number 2
  test("updates task data when modified", async () => {
    // Create a modified version of the mock tasks
    const modifiedTasks = [
      {
        id: 1,
        title: "Updated Task 1", // Changed task name
        description: "Description for Task 1",
        estimatedTime: 8, // Changed estimated hours
        done: 0, // Status remains the same
        story_Points: 5, // Changed story points
        moduleId: 1,
        responsible: 2, // Changed developer
        actualTime: 0,
      },
      {
        id: 2,
        title: "Task 2",
        description: "Description for Task 2",
        estimatedTime: 8,
        done: 0,
        story_Points: 5,
        moduleId: 2,
        responsible: 2,
        actualTime: 0,
      },
      {
        id: 3,
        title: "Task 3",
        description: "Description for Task 3",
        estimatedTime: 3,
        done: 1,
        story_Points: 2,
        moduleId: 1,
        responsible: 1,
        actualTime: 4,
      },
    ];

    // Mock the GET request for tasks with the modified data
    server.use(
      rest.get(`${API_LIST}`, (req, res, ctx) => {
        return res(ctx.status(200), ctx.json(modifiedTasks));
      })
    );

    // Mock the PUT request for updating a task
    server.use(
      rest.put(`${API_LIST}/1`, (req, res, ctx) => {
        return res(
          ctx.status(200),
          ctx.json({
            id: 1,
            title: "Updated Task 1",
            description: "Description for Task 1",
            estimatedTime: 8,
            done: 0, // Status remains the same
            story_Points: 5,
            moduleId: 1,
            responsible: 2,
            actualTime: 0,
          })
        );
      })
    );

    render(<Dashboard />);

    // Wait for the component to load
    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    // Find the task by its ID in the table
    const taskRows = screen.getAllByRole("row");
    const taskRow = taskRows.find((row) =>
      row.textContent.includes("Updated Task 1")
    );

    // Verify the task data has been updated
    expect(taskRow).toBeInTheDocument();
    expect(taskRow).toHaveTextContent("user2"); // Updated developer
    expect(taskRow).toHaveTextContent("5"); // Updated story points
    expect(taskRow).toHaveTextContent("8"); // Updated estimated hours
  });

  // Test listing completed tasks by sprint
  test("lists completed tasks by sprint with minimum information", async () => {
    render(<Dashboard />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    // Find the filter select element
    const filterContainer = screen
      .getByText("Filter by Sprint:")
      .closest("div");
    const moduleSelect = filterContainer.querySelector("select");

    // Select Sprint 1
    fireEvent.change(moduleSelect, { target: { value: "1" } });

    // Wait for the UI to update
    await waitFor(() => {
      // Check completed task in Sprint 1
      const completedTask = screen.getByText("Task 3").closest("tr");
      expect(completedTask).toHaveTextContent("user1"); // Developer name
      expect(completedTask).toHaveTextContent("3"); // Estimated hours
      expect(completedTask).toHaveTextContent("4"); // Actual hours
    });

    // Select Sprint 2
    fireEvent.change(moduleSelect, { target: { value: "2" } });

    // Wait for the UI to update
    await waitFor(() => {
      // No completed tasks in Sprint 2
      expect(screen.queryByText("Task 3")).not.toBeInTheDocument();
    });
  });
});
