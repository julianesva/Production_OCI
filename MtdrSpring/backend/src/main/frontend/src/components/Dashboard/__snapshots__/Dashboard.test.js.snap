// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component Dashboard component matches snapshot 1`] = `
<div>
  <div
    class="dashboard-main"
  >
    <div
      class="dashboard-main-container"
    >
      <div
        class="dashboard-main-content"
      >
        <div
          class="dashboard-main-content-container"
        >
          <div
            class="dashboard-title-container"
          >
            <p
              class="dashboard-title-text"
            >
              Dashboard
            </p>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11n98c0-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Recommendation
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="dashboard-input-container"
          >
            <input
              class="dashboard-input-format"
              placeholder="Title"
              type="text"
              value=""
            />
            <input
              class="dashboard-input-format description-input"
              placeholder="Description"
              type="text"
              value=""
            />
            <select
              class="dashboard-input-format dashboard-module-select-input"
            >
              <option
                disabled=""
                value=""
              >
                Responsible
              </option>
              <option
                value="1"
              >
                user1
              </option>
              <option
                value="2"
              >
                user2
              </option>
            </select>
            <input
              class="dashboard-input-format num-input"
              max="10"
              min="1"
              placeholder="Story Points"
              type="number"
              value=""
            />
            <input
              class="dashboard-input-format num-input"
              min="1"
              placeholder="Hours"
              type="number"
              value=""
            />
            <select
              class="dashboard-input-format dashboard-module-select-input"
              data-testid="module-select"
              id="moduleSelect"
            >
              <option
                disabled=""
                value=""
              >
                Sprint
              </option>
            </select>
            <button
              class="dashboard-input-button"
            >
              <p
                class="dashboard-input-button-text"
              >
                Add
              </p>
            </button>
          </div>
          <div
            class="filter-module-container"
          >
            <p
              class="filter-module-title-text"
            >
              Filter by Sprint:
            </p>
            <select
              class="filter-module-select"
              data-testid="filter-module-select"
            >
              <option
                value="all"
              >
                All
              </option>
              <option
                value="1"
              >
                1 - Sprint 1
              </option>
              <option
                value="2"
              >
                2 - Sprint 2
              </option>
            </select>
          </div>
          <div
            class="dashboard-table-container"
          >
            <div
              class="dashboard-table-title-container"
            >
              <p
                class="dashboard-table-title"
              >
                To Do
              </p>
              <button>
                <svg
                  class="tabler-icon tabler-icon-caret-down "
                  fill="none"
                  height="25px"
                  stroke="black"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="25px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 10l6 6l6 -6h-12"
                  />
                </svg>
              </button>
            </div>
            <div
              class="dashboard-table-big-separation"
            >
              <div
                class="dashboard-table-separation"
              >
                <table
                  class="dashboard-table-task-table"
                >
                  <thead>
                    <tr>
                      <th
                        class="dashboard-table-task-table-head-left"
                      >
                        Title
                      </th>
                      <th
                        class="dashboard-table-task-table-head-left dashboard-table-border-inline"
                      >
                        Description
                      </th>
                      <th
                        class="dashboard-table-task-table-head-left dashboard-table-border-inline"
                      >
                        Responsible
                      </th>
                      <th
                        class="dashboard-table-task-table-head-center dashboard-table-border-inline"
                      >
                        Hours
                      </th>
                      <th
                        class="dashboard-table-task-table-head-center dashboard-table-border-inline"
                      >
                        Story Points
                      </th>
                      <th
                        class="dashboard-table-task-table-head-actions"
                      >
                        Actions
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr
                      class="dashboard-table-row"
                    >
                      <td
                        class="dashboard-table-text-column dashboard-table-title-column"
                      >
                        Task 1
                      </td>
                      <td
                        class="dashboard-table-text-column dashboard-table-description-column dashboard-table-border-inline"
                      >
                        Description for Task 1
                      </td>
                      <td
                        class="dashboard-table-text-column dashboard-table-responsible-column dashboard-table-border-inline"
                      >
                        user1
                      </td>
                      <td
                        class="dashboard-table-num-column dashboard-table-border-inline"
                      >
                        5
                      </td>
                      <td
                        class="dashboard-table-num-column dashboard-table-border-inline"
                      >
                        3
                      </td>
                      <td
                        class="dashboard-table-actions-column"
                      >
                        <div
                          class="dashboard-table-actions-container"
                        >
                          <button
                            class="dashboard-table-action-next-button"
                          >
                            Done
                          </button>
                          <button
                            class="dashboard-table-action-trash-button"
                          >
                            <svg
                              class="tabler-icon tabler-icon-trash "
                              fill="none"
                              height="16px"
                              stroke="white"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                              viewBox="0 0 24 24"
                              width="16px"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M4 7l16 0"
                              />
                              <path
                                d="M10 11l0 6"
                              />
                              <path
                                d="M14 11l0 6"
                              />
                              <path
                                d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
                              />
                              <path
                                d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
                              />
                            </svg>
                          </button>
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="dashboard-table-row"
                    >
                      <td
                        class="dashboard-table-text-column dashboard-table-title-column"
                      >
                        Task 2
                      </td>
                      <td
                        class="dashboard-table-text-column dashboard-table-description-column dashboard-table-border-inline"
                      >
                        Description for Task 2
                      </td>
                      <td
                        class="dashboard-table-text-column dashboard-table-responsible-column dashboard-table-border-inline"
                      >
                        user2
                      </td>
                      <td
                        class="dashboard-table-num-column dashboard-table-border-inline"
                      >
                        8
                      </td>
                      <td
                        class="dashboard-table-num-column dashboard-table-border-inline"
                      >
                        5
                      </td>
                      <td
                        class="dashboard-table-actions-column"
                      >
                        <div
                          class="dashboard-table-actions-container"
                        >
                          <button
                            class="dashboard-table-action-next-button"
                          >
                            Done
                          </button>
                          <button
                            class="dashboard-table-action-trash-button"
                          >
                            <svg
                              class="tabler-icon tabler-icon-trash "
                              fill="none"
                              height="16px"
                              stroke="white"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                              viewBox="0 0 24 24"
                              width="16px"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M4 7l16 0"
                              />
                              <path
                                d="M10 11l0 6"
                              />
                              <path
                                d="M14 11l0 6"
                              />
                              <path
                                d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
                              />
                              <path
                                d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
                              />
                            </svg>
                          </button>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <div
            class="dashboard-table-container"
          >
            <div
              class="dashboard-table-title-container"
            >
              <p
                class="dashboard-table-title"
              >
                Completed
              </p>
              <button>
                <svg
                  class="tabler-icon tabler-icon-caret-down "
                  fill="none"
                  height="25px"
                  stroke="black"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="25px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 10l6 6l6 -6h-12"
                  />
                </svg>
              </button>
            </div>
            <div
              class="dashboard-table-big-separation"
            >
              <div
                class="dashboard-table-separation"
              >
                <table
                  class="dashboard-table-task-table"
                >
                  <thead>
                    <tr>
                      <th
                        class="dashboard-table-task-table-head-left"
                      >
                        Title
                      </th>
                      <th
                        class="dashboard-table-task-table-head-left dashboard-table-border-inline"
                      >
                        Description
                      </th>
                      <th
                        class="dashboard-table-task-table-head-left dashboard-table-border-inline"
                      >
                        Responsible
                      </th>
                      <th
                        class="dashboard-table-task-table-head-center dashboard-table-border-inline"
                      >
                        Hours
                      </th>
                      <th
                        class="dashboard-table-task-table-head-center dashboard-table-border-inline"
                      >
                        Real Hours
                      </th>
                      <th
                        class="dashboard-table-task-table-head-center dashboard-table-border-inline"
                      >
                        Story Points
                      </th>
                      <th
                        class="dashboard-table-task-table-head-actions"
                      >
                        Actions
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr
                      class="dashboard-table-row"
                    >
                      <td
                        class="dashboard-table-text-column dashboard-table-title-column"
                      >
                        Task 3
                      </td>
                      <td
                        class="dashboard-table-text-column dashboard-table-description-column dashboard-table-border-inline"
                      >
                        Description for Task 3
                      </td>
                      <td
                        class="dashboard-table-text-column dashboard-table-responsible-column dashboard-table-border-inline"
                      >
                        user1
                      </td>
                      <td
                        class="dashboard-table-num-column dashboard-table-border-inline"
                      >
                        3
                      </td>
                      <td
                        class="dashboard-table-num-column dashboard-table-border-inline"
                      >
                        4
                      </td>
                      <td
                        class="dashboard-table-num-column dashboard-table-border-inline"
                      >
                        2
                      </td>
                      <td
                        class="dashboard-table-actions-column"
                      >
                        <div
                          class="dashboard-table-actions-container"
                        >
                          <button
                            class="dashboard-table-action-next-button"
                          >
                            Undo
                          </button>
                          <button
                            class="dashboard-table-action-trash-button"
                          >
                            <svg
                              class="tabler-icon tabler-icon-trash "
                              fill="none"
                              height="16px"
                              stroke="white"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                              viewBox="0 0 24 24"
                              width="16px"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M4 7l16 0"
                              />
                              <path
                                d="M10 11l0 6"
                              />
                              <path
                                d="M14 11l0 6"
                              />
                              <path
                                d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
                              />
                              <path
                                d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
                              />
                            </svg>
                          </button>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
